allow no counts from dedup

rs-tohkubo opened the pull request:
-deleted

rs-mkaiser commented on tap_s3_csv/conversion.py:
```diff
@@ -111,6 +111,9 @@ def generate_schema(samples):
     counts = count_samples(samples)
 
+    if not counts:
+        return to_return
```
okay so if generate schema returns here, won't the code relying on the schema then fail? Have you tested it?

rs-mkaiser replied:
My question is still not addressed

rs-tohkubo replied:
the case we're trying to handle is when counts is None. The unit test I added handles that case.

rs-mkaiser replied:
**won't the code relying on the schema then fail?**

rs-mkaiser commented:
no tests? How do you know this change fixes anything?

rs-mkaiser reviewed (changes requested):
need either really convincing answers to my questions that shows that this will fix the problem, or some tests

rs-tohkubo commented:
Added a test to make sure that when no samples are received after dedup, the counts dictionary won't be populated and added a check to make sure it doesn't break when counts returns an empty dict.

rs-tohkubo commented:
Upon further investigation, it ended up being a simple version bump of singer docker component (was using an outdated version).
